controllers/storagecluster/cephrbdmirrors.go

```diff
@@ -124,3 +134,210 @@ func (obj *ocsCephRbdMirrors) ensureDeleted(r *StorageClusterReconciler, sc *ocs
 	}
 	return reconcile.Result{}, r.deleteCephRbdMirrorInstance(cephRbdMirrors)
 }
+
+func (r *StorageClusterReconciler) ensureRbdMirrorDebugLogging(sc *ocsv1.StorageCluster) error {
+	enableRbdMirrorDebugLoggingJobName := "enable-rbd-mirror-debug-logging"
+	disableRbdMirrorDebugLoggingJobName := "disable-rbd-mirror-debug-logging"
+	enableRbdMirrorDebugLoggingCommands :=
+		`echo "starting configuration of mirror logging"
+ceph config set client.rbd-mirror.a debug_ms 1
+ceph config set client.rbd-mirror.a debug_rbd 15
+ceph config set client.rbd-mirror.a debug_rbd_mirror 30
+ceph config set client.rbd-mirror.a log_file /var/log/ceph/\$cluster-\$name.log
+ceph config set client.rbd-mirror-peer debug_ms 1
+ceph config set client.rbd-mirror-peer debug_rbd 15
+ceph config set client.rbd-mirror-peer debug_rbd_mirror 30
+ceph config set client.rbd-mirror-peer log_file /var/log/ceph/\$cluster-\$name.log
+ceph config set mgr mgr/rbd_support/log_level debug
+		echo "completed"`
+	disableRbdMirrorDebugLoggingCommands :=
+		`echo "Removing configuration of mirror logging"
+ceph config rm client.rbd-mirror.a debug_ms
+ceph config rm client.rbd-mirror.a debug_rbd
+ceph config rm client.rbd-mirror.a debug_rbd_mirror
+ceph config rm client.rbd-mirror.a log_file
+ceph config rm client.rbd-mirror-peer debug_ms
+ceph config rm client.rbd-mirror-peer debug_rbd
+ceph config rm client.rbd-mirror-peer debug_rbd_mirror
+ceph config rm client.rbd-mirror-peer log_file
+ceph config rm mgr mgr/rbd_support/log_level
+		echo "completed"`
+
+	// Mirrooring is enabled but the debug logging is not enabled
+	if sc.Spec.Mirroring.Enabled && !rbdMirrorDebugLoggingEnabled {
+
+		err := r.Client.Create(r.ctx, getRbdMirrorDebugLoggingJob(sc, enableRbdMirrorDebugLoggingJobName, enableRbdMirrorDebugLoggingCommands))
+		if err != nil && !errors.IsAlreadyExists(err) {
+			return err
+		}
+		// Wait for the job to succeed
+		err = r.waitForJobCompletion(enableRbdMirrorDebugLoggingJobName, sc.Namespace)
```

Comment:
I don't think it is a good idea to wait in the reconcile, instead requeue after sometime

Reply:
Yes, Will requeue instead of wait